keps/0004-add-testing-infrastructure.md

```diff
@@ -84,7 +84,41 @@ exists in real world situations.
 
 ### CICD
 
-Identify and implement a CICD toolkit that can be integrated into GitHub to validate all PRs.
+Use [CircleCI](https://circleci.com/docs/) and the [GitHub Plugin](https://github.com/marketplace/circleci/plan/MDIyOk1hcmtldHBsYWNlTGlzdGluZ1BsYW45MA==#pricing-and-setup).
+For OpenSource projects we will recieve 1,000 monthly build minutes.  With the test suite below, that should suffice as a baseline.
+
+#### Pull Requests
+
+All Pull Requests into master need to have the following checks pass.  These should be ordered in fastest to slowest to reduce the time spent when/if failures occur
+
+0) Check author has signed CLA
+1) `go fmt` does not change anything 
+2) `go lint` and `go vet` both pass
+3) All unit tests pass (with `-race` flag)
+4) Dockerfile builds (this requires all dependencies in the vendor folder)
+
```

Comment:
Thinking about this should we add `5) Integration tests pass` as talked about under [Tests](https://github.com/kudobuilder/kudo/blob/0600848c755476c6e250b904477d5608db60812f/keps/0004-add-testing-infrastructure.md#tests) which touches also this [comment](https://github.com/kudobuilder/kudo/pull/144/files#r263640425) ?

For that we would also need to add probably a way to test on a running Kubernetes cluster, which is currently marked as optional when looking at different infrastructure for pipelines needed ([here](https://github.com/kudobuilder/kudo/blob/0600848c755476c6e250b904477d5608db60812f/keps/0004-add-testing-infrastructure.md#infrastructure-needed-optional)). In general, e.g. for the CLI we would need similar like in the [Kubernetes Command-Line Integration Test Suite](https://github.com/kubernetes/kubernetes/tree/master/test/cmd) and [here](https://github.com/kubernetes/kubernetes/blob/master/hack/lib/test.sh) a place for those tests to live. Plus they need the proper infrastructure to run in.